Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -206,6 +207,53 @@ await container.UpsertItemAsync<TItem>(value, new PartitionKey(value.PartitionKe
             TryLogDebugDetails(_logger, () => $"Deleted: {id}");
         }
 
+        /// <inheritdoc/>
+        public ValueTask<bool> ExistsAsync(string id, string partitionKeyValue = null, CancellationToken cancellationToken = default)
+            => ExistsAsync(id, new PartitionKey(partitionKeyValue ?? id), cancellationToken);
+
+        /// <inheritdoc/>
+        public async ValueTask<bool> ExistsAsync(string id, PartitionKey partitionKey, CancellationToken cancellationToken = default)
+        {
+            Container container =
+                await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
+
+            try
+            {
+                ItemResponse<TItem> response =

```

Comment:
Should we be checking that the `response.Value`? Is it possible that the `response` is `null` or that the `.Value` is missing?

Reply:
I have an extension method to support this currently on our project, as far as I have seen so far is when the item does not exists a 404 is thrown almost every time.

Reply:
I believe that's the documented behavior of Cosmos: https://docs.microsoft.com/en-us/rest/api/cosmos-db/get-a-document#status-codes

Reply:
Yes think you are correct.

Reply:
Perfect, thanks